smarts/sstudio/types.py

```diff
@@ -813,6 +813,36 @@ def to_geometry(self, road_map: Optional[RoadMap] = None) -> Polygon:
             poly = shapely_rotate(poly, self.rotation, use_radians=True)
         return shapely_translate(poly, xoff=x, yoff=y)
 
+@dataclass(frozen=True)
+class ConfigurableZone(Zone):
+    """A descripter that defines a specific configurableZone defined by user"""
+
+    ext_coordinates: List[Tuple[float, float]]
+    """external coordinates of the polygon"""
+    rotation: Optional[float] = None
+    """The heading direction of the bubble(radians, clock-wise rotation)"""
+
+    def to_geometry(self, road_map: Optional[RoadMap] = None) -> Polygon:
+        """Generate a polygon according to given coordinates"""
+        poly = None
+        if (
+            len(self.ext_coordinates) == 2
+        ):  # if user only specified two points, create a box
+            x_min = min(self.ext_coordinates[0][0], self.ext_coordinates[1][0])
+            x_max = max(self.ext_coordinates[0][0], self.ext_coordinates[1][0])
+            y_min = min(self.ext_coordinates[1][0], self.ext_coordinates[1][1])
+            y_max = max(self.ext_coordinates[1][0], self.ext_coordinates[1][1])
+            poly = Polygon(
+                [(x_min, y_min), (x_max, y_min), (x_max, y_max), (x_min, y_max)],
+            )
```

Comment:
Will this cause an error if you provide a situation where it describes an axis-parallel line: `p1.x == p2.x` or `p1.y == p2.y`?

The box would have a width or height of 0 then.

Reply:
Maybe I could add the check for this in __post__init__ together with the point numbers check